crud/readview.lua

```diff
@@ -0,0 +1,613 @@
+local errors = require('errors')
+
+local fiber = require('fiber')
+local common = require('crud.select.compat.common')
+
+local Merger = require('crud.select.merger')
+
+local stash = require('crud.common.stash')
+local utils = require('crud.common.utils')
+local sharding = require('crud.common.sharding')
+local select_executor = require('crud.select.executor')
+local select_filters = require('crud.compare.filters')
+local dev_checks = require('crud.common.dev_checks')
+local schema = require('crud.common.schema')
+local const = require('crud.common.const')
+local sharding_metadata_module = require('crud.common.sharding.sharding_metadata')
+local stats = require('crud.stats')
+local checks = require('checks')
+
+local compare_conditions = require('crud.compare.conditions')
+local select_plan = require('crud.compare.plan')
+
+local ReadviewError = errors.new_class('ReadviewError', {capture_stack = false})
+
+local readview = {}
+
+function checkers.vshard_call_mode(p)
+    return p == 'write' or p == 'read'
+end
+
+local function readview_open_on_storage(readview_name)
+    local read_view = box.read_view.open({name = readview_name})
+
+    if read_view == nil then
+        return "error creating readview"
+    end
+
+    local result = {}
+    result.uuid = box.info().uuid
+    result.id = read_view.id
+
+    return result
+end
+
+local function readview_close_on_storage(readview_uuid)
+    dev_checks('table')
+
+    local list = box.read_view.list()
+
+    local readview_id
+    for _, replica_info in pairs(readview_uuid) do
+        if replica_info.uuid == box.info().uuid then
+            readview_id = replica_info.id
+        end
+    end
+
+    if readview_id == nil then
+        return ""
+    end
+
+    for k,v in pairs(list) do
+        if v.id == readview_id then
+            list[k]:close()
+            return list[k]:info()
+        end
+    end
+
+    return "no such readview"
```

Comment:
It would also be nice to implement at least one test to catch these error. For example, manually close a readview on storage and see what will happen.

Reply:
added